### PR TITLE
Science-Related Jobs gets Mood Bonus for Petting Cyborgs

### PR DESCRIPTION
**Pull request by Runian**
# Document the changes in your pull request
Port of tgstation/tgstation/pull/46296.
The science department (Scientist, Roboticist, and Research Director) and any adjacent jobs (Geneticist) can pet cyborgs to get an increase to their mood.

Mood events can be restricted to certain jobs and will only allow these certain jobs to get the mood event.
Removes `COMSIG_ADD_MOOD_EVENT_RND` since it is unused.

# Changelog
:cl:  
rscadd: Mood events can be restricted to jobs.
rscadd: Petting a Cyborg is now a positive mood event for all science members (and the geneticist too).
/:cl:

### CODE REVIEW

**Comment by Therandomhoboo** [comment]
WE should give a mood buff to everyone else that kills the cyborg that ISN'T in the sci department.

File: code/datums/components/mood.dm [hunk 1]
```diff
@@ -28,6 +28,11 @@
 		modify_hud()
 		var/datum/hud/hud = owner.hud_used
 		hud.show_hud(hud.hud_version)
+	if(!ishuman(owner))
+		return
+	var/mob/living/carbon/human/H = owner
+	if(H.mind?.assigned_role in list("Research Director", "Scientist", "Roboticist"))
```
Comment (JohnFulpWillard) [inline code/datums/components/mood.dm]: Gene should be included since they are part of science somewhat

Comment (Runian) [reply code/datums/components/mood.dm]: done

**Review by Moltijoe — COMMENTED** [review]
Are there going to be enough RND mood events that they need their own signal?

**Comment by Runian** [comment]
> Are there going to be enough RND mood events that they need their own signal?

This is the only mood event I plan to add. Unless someone plans to port over tgstation/tgstation/pull/68592 or has a better way that isn't checking if the mob uses mood in `silicon_defense`, I think that having its own signal is the best way of doing all of this even if it might be only the mood event that uses the signal.

**Comment by Runian** [comment]
> Are there going to be enough RND mood events that they need their own signal?

`COMSIG_ADD_MOOD_EVENT_RND` was added by #17260 and was entirely unused. Anyways, changed how all of this is done with a list in the mood event and removed `COMSIG_ADD_MOOD_EVENT_RND` too.

**Review by JohnFulpWillard — APPROVED** [review]
we need to port tg mood system already

**Comment by GDRdev** [comment]
tcoms should be able to pet the AI core and get a mood bonus from that as well

**Comment by Runian** [comment]
> tcoms should be able to pet the AI core and get a mood bonus from that as well

If they become a more science dedicated role, sure (#18568).

**Pull request merged**